File: demo/src/main/java/com/example/demo/project/spring/entities/City.java
package com.example.demo.project.spring.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "CITIES", uniqueConstraints = {
        @UniqueConstraint(columnNames = {"name","postal_code"})})
public class City {

    @Id
    @GeneratedValue(generator = "gen")
    @SequenceGenerator(name = "gen", sequenceName = "city_seq")
    @Column(name = "id")
    private int id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "country", nullable = false)
    private String country;

    @Column(name = "region")
    private String region;

    @Column(name = "postal_code")
    private String postalCode;

    @JsonManagedReference
    @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL, mappedBy = "birthplace")
    private List<Person> registeredPeople;

    @JsonBackReference
    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinColumn(name = "issuingAuthority_id")
    private IssuingAuthority issuingAuthority;

    public City() {}

    public City(String name, String country, String region, String postalCode) {
        this.name = name;
        this.country = country;
        this.region = region;
        this.postalCode = postalCode;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getRegion() {
        return region;
    }

    public void setRegion(String region) {
        this.region = region;
    }

    public String getPostalCode() {
        return postalCode;
    }

    public void setPostalCode(String postalCode) {
        this.postalCode = postalCode;
    }

    public List<Person> getRegisteredPeople() {
        return registeredPeople;
    }

    public void setRegisteredPeople(List<Person> registeredPeople) {
        this.registeredPeople = registeredPeople;
    }

    public IssuingAuthority getIssuingAuthority() {
        return issuingAuthority;
    }

    public void setIssuingAuthority(IssuingAuthority issuingAuthority) {
        this.issuingAuthority = issuingAuthority;
    }
}

File: angular/src/app/model/city.ts
import {Person} from "./person";
import {IssuingAuthority} from "./issuing-authority";

export class City {
  private _id: number
  private _name: String
  private _country: String
  private _region: String
  private _postalCode: String
  private _registeredPeople: Person[]
  private _issuingAuthority: IssuingAuthority

  constructor (obj?: any) {
    this._id = (obj && obj._id);
    this._name = (obj && obj._name);
    this._country = (obj && obj._country);
    this._region = (obj && obj._region);
    this._postalCode = (obj && obj._postalCode);
    this._registeredPeople = (obj && obj._registeredPeople);
    this._issuingAuthority = (obj && obj._issuingAuthority);
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get name(): String {
    return this._name;
  }

  set name(value: String) {
    this._name = value;
  }

  get country(): String {
    return this._country;
  }

  set country(value: String) {
    this._country = value;
  }

  get region(): String {
    return this._region;
  }

  set region(value: String) {
    this._region = value;
  }

  get postalCode(): String {
    return this._postalCode;
  }

  set postalCode(value: String) {
    this._postalCode = value;
  }

  get registeredPeople(): Person[] {
    return this._registeredPeople;
  }

  set registeredPeople(value: Person[]) {
    this._registeredPeople = value;
  }

  get issuingAuthority(): IssuingAuthority {
    return this._issuingAuthority;
  }

  set issuingAuthority(value: IssuingAuthority) {
    this._issuingAuthority = value;
  }
}
